cmd/server/cadence/server.go

```diff
@@ -220,7 +223,20 @@ func (s *server) startService() common.Daemon {
 			log.Fatalf("not able to find advanced visibility store in config: %v", advancedVisStoreKey)
 		}
 
-		params.ESConfig = advancedVisStore.ElasticSearch
+		if params.PersistenceConfig.AdvancedVisibilityStore == common.PinotVisibilityStoreName {
+			params.PinotConfig = advancedVisStore.Pinot
+			esVisStore := s.cfg.Persistence.DataStores[common.ESVisibilityStoreName]
```

Comment:
why would this be defined?

Reply:
We need esVisStore to write to ES, we have components like ESAnalyzer is using ElasticSearch. The plan is to clean those after we switch to operate on Pinot.